src/setup_payload/AdditionalDataPayloadParser.cpp

```diff
@@ -42,20 +43,27 @@ CHIP_ERROR AdditionalDataPayloadParser::populatePayload(SetupPayloadData::Additi
     TLV::ContiguousBufferTLVReader innerReader;
 
     reader.Init(mPayloadBufferData, mPayloadBufferLength);
-    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag));
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, RotatingDeviceId::DiscoveryExtensionDescriptorTag));
 
     // Open the container
     ReturnErrorOnFailure(reader.OpenContainer(innerReader));
+    if (innerReader.Next(TLV::kTLVType_ByteString, TLV::ContextTag(SetupPayloadData::kRotatingDeviceIdTag)) == CHIP_NO_ERROR)
+    {
+        // Get the value of the rotating device id
+        ByteSpan rotatingDeviceId(innerReader.GetReadPoint(), innerReader.GetLength());
 
-    ReturnErrorOnFailure(innerReader.Next(TLV::kTLVType_UTF8String, TLV::ContextTag(SetupPayloadData::kRotatingDeviceIdTag)));
+        // This test uses <, not <=, because kHexMaxLength includes the null-terminator.
+        VerifyOrReturnError(rotatingDeviceId.size() < RotatingDeviceId::kHexMaxLength, CHIP_ERROR_INVALID_STRING_LENGTH);
```

Comment:
This test makes no sense in this form, because now we have a _binary_ rotating id, so why are we comparing it to hex max length?

Reply:
agree , updated the condition